cmd/postpolicyform.go

```diff
@@ -311,6 +345,15 @@ func checkPostPolicy(formValues http.Header, postPolicyForm PostPolicyForm) erro
 				return fmt.Errorf("Invalid according to Policy: Policy Condition failed: [%s, %s, %s]", op, policy.Key, policy.Value)
 			}
 		}
+		delete(checkHeader, formCanonicalName)
+	}
+
+	if len(checkHeader) != 0 {
+		logKeys := make([]string, 0, len(checkHeader))
+		for key := range checkHeader {
+			logKeys = append(logKeys, key)
+		}
+		return fmt.Errorf("Each form field that you specify in a form (except %s) must appear in the list of conditions.", strings.Join(logKeys, ", "))
```

Comment:
Are you sure? I recently hit that error message, and i had to add the keys listed in the "except" list to get it working again.

Reply:
Not all fields you need to provide what you upload with standard headers. All in the name of AWS S3 compatibility unfortunately.

Reply:
Yes I agree with you but the wording gave me troubles. The word "except" made me think all the fields must be included excepted the one listed in the parenthesis. Maybe just remove that word?

Reply:
I'm still lost. I keep getting `Access Denied. (Each form field that you specify in a form (except Signature, Awsaccesskeyid) must appear in the list of conditions`.

Reply:
Yeah. You can add them to policy like
```go
        policy.SetKey(file.Name())
	policy.SetContentType("image/png")
``` .  Or add the `x-amz-ignore-Signature` or `x-amz-ignore-Awsaccesskeyid` to postfrom. What kind of code do you use? Go or Java or others. Maybe I can found something for you. @horta